### PR TITLE
Add all commands supported by LeoFS 1.4.1

### PR DESCRIPTION
**Pull request by mocchira**
The commits being included in this PR fall into the three categories.

- Update dependencies
  - [Bump jsx to the 2.9.0](https://github.com/project-fifo/libleofs/commit/2d41b83ec70707ca4296fdde5bf87ab30d4bd6ce)
- Bug fixes
  - [Add the timeout to gen_tcp:(connect|recv) to avoid the deadlock](https://github.com/project-fifo/libleofs/commit/589ad7b1714de6c0244ab202c4677ff7784ac3d7)
  - [Make cmd/3 receive a complete json body even if a response is split into small pieces](https://github.com/project-fifo/libleofs/commit/473f8cf0071c0eccce22893ec09b7bf8ff8acac4)
- Add new functions
  - [Add rollback/3 and mq related functions](https://github.com/project-fifo/libleofs/commit/5bf650ac04df63e114034f5429a3945892418399)
  - [Add recover_(disk|consistency)](https://github.com/project-fifo/libleofs/commit/67df2108126b3ddddcf391a5bd31ad7a9740a693)
  - [Add diagnose_start](https://github.com/project-fifo/libleofs/commit/1a431f102b96bc10b6d9ad19bb054abb60e18ea9)
  - [Add manager maintenance related functions](https://github.com/project-fifo/libleofs/commit/2c963b48f5765baf8da1faaff7524cdc1088464a)

### CODE REVIEW

**Comment by Licenser** [comment]
Hi,
I just wanted to say that I saw the pr and have not forgotten it but am
travelling until next week and won't have time to review and merge it until
then

**Comment by mocchira** [comment]
@Licenser Got it. Have a great travel :)

File: rebar.config [hunk 1]
```diff
@@ -4,5 +4,5 @@
 {deps,
  [
   {lager, ".*", {git, "git://github.com/basho/lager.git", {tag, "2.0.3"}}},
-  {jsx, "1.4.5"}
+  {jsx,   ".*", {git, "git://github.com/talentdeficit/jsx.git", {tag, "2.9.0"}}}
```
Comment (Licenser) [inline rebar.config]: I would prefer this to remain a hex dependency, in fact both should be (to publish to hex) but the 1st one is on me

**Review by Licenser — CHANGES_REQUESTED** [review]

**Comment by Licenser** [comment]
Sorry for the delay :) back in the country now only a tiny thing to change the rest looks really good and a big improvement over the original code! Thanks!

File: rebar.config (hunk 1 above)
Comment (mocchira) [reply rebar.config]: @Licenser Got it. let me ask you that should I also patch Makefile and include rebar3 in libleofs in order to resolve a hex dependency like the following?

```diff
diff --git a/Makefile b/Makefile
index 0217c27..60787cf 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-REBAR = $(shell pwd)/rebar
+REBAR = $(shell pwd)/rebar3

 .PHONY: deps rel package

diff --git a/rebar b/rebar
deleted file mode 100755
index b0781c0..0000000
Binary files a/rebar and /dev/null differ
diff --git a/rebar.config b/rebar.config
index e67b89f..ca0f7d3 100644
--- a/rebar.config
+++ b/rebar.config
@@ -4,5 +4,5 @@
 {deps,
  [
   {lager, ".*", {git, "git://github.com/basho/lager.git", {tag, "2.0.3"}}},
-  {jsx,   ".*", {git, "git://github.com/talentdeficit/jsx.git", {tag, "2.9.0"}}}
+  {jsx, "2.9.0"}
  ]}.
```

Comment (Licenser) [reply rebar.config]: looks good to me!

**Comment by mocchira** [comment]
@Licenser Done. Please take another look.

**Review by Licenser — APPROVED** [review]
awesome!

**Comment by Licenser** [comment]
Thanks a lot @mocchira ! I've merged it all, moved lager to hex and published it as `v0.1.3` both as tag adn on hex :)

**Pull request merged**